README.md
# Vue3-Uni-TS-Template基础模板

## [官方文档：https://fant-mini-plus.top](https://fant-mini-plus.top/fant-mini-plus/)

## [开发计划（规划中）](https://docs.qq.com/sheet/DSVZOdkhOalFPd2NZ?tab=csws7k&u=60ccbc87e7b04883b8c01ab07165a577)

关于开发计划的问题或者建议请先到[Gitee](https://gitee.com/fant-mini/uniapp-vue3-fant-ts)或者[Github](https://github.com/Moonofweisheng/uniapp-vue3-fant-ts)提出issue，或者在评论区发表评论，方便我记录问题以及安排后续的开发计划。（组件或者模板使用上有什么问题，希望增加什么组件或者模板页面都可以提）

#### 介绍
基于uni-app和fant-mini-plus的vue3项目基础模板，使用vue3、vite4、[fant-mini-plus](https://ext.dcloud.net.cn/plugin?id=11489)、pinia、[uni-mini-router](https://ext.dcloud.net.cn/plugin?id=11208)、axios、[fant-axios-adapter](https://ext.dcloud.net.cn/plugin?id=11817)开发，基于koa实现mock功能

#### 特性

- 🚀 使用vue3/vite4/pinia/axios等主流框架开发
- 🚀 引入fant-mini-plus组件库，开箱即用
- 💪 引入Eslint/Prettier，统一前端代码风格
- 💪 引入lint-staged/husky/commitlint/commitizen/standard-version，统一代码提交规范，自动生成版本发布日志
- 💪 使用pinia，并支持持久化存储
- 💪 提供koa-mock本地mock服务
- 💪 支持axios取消请求（基于requestTask.abort()实现）
## 注意！！！
>插件市场会将类似`.eslintrc.js`这种以`.`开头的文件忽略掉，故我将`.`改成了`_`上传。如果直接在插件市场下载，需要手动将`_`改为`.`，涉及文件及目录包含`_husky`、`_vscode`、`_env.development`、`_env.production`、`_eslintignore`、`_eslintrc.js`、`_git-cz.json`、`_gitignore`、`_prettierrc`。或者到[gitee](https://gitee.com/fant-mini/uniapp-vue3-fant-ts)上`clone`或`fork`


#### 项目依赖

| 功能 | 依赖库 |
| --- | --- |
| 组件库 | [fant-mini-plus](https://ext.dcloud.net.cn/plugin?id=11489) |
| 路由（name跳转和导航守卫） | [uni-mini-router](https://ext.dcloud.net.cn/plugin?id=11208) |
| 网络请求（支持取消请求） | `axios`  [fant-axios-adapter](https://ext.dcloud.net.cn/plugin?id=11817) |
| 代码规范 | `Eslint` `Prettier` |
| 提交规范 | `lint-staged` `husky` `commitlint` `commitizen` |
| 发布日志 | `standard-version` |
| 状态管理 | `pinia`（自定义插件实现持久化） |

#### 项目结构
```shell
uniapp-vue3-fant-ts
├─ .env.development----------------------------------环境变量文件dev时生效，VITE_BASEURL为接口地址
├─ .env.production-----------------------------------环境变量文件build时生效，VITE_BASEURL为接口地址
├─ .eslintignore-------------------------------------eslint豁免校验的配置，配置的文件免于eslint校验
├─ .eslintrc.js--------------------------------------eslint配置，指定eslint校验规则
├─ .gitignore
├─ .husky--------------------------------------------git hooks，此处配置了pre-commit和commit-msg两个
├─ .prettierrc---------------------------------------prettier配置，指定prettier校验规则
├─ .vscode-------------------------------------------vscode配置，实现保存的时候格式化
├─ CHANGELOG.md--------------------------------------版本发布日志
├─ README.md
├─ commitlint.config.js------------------------------commitlint配置文件
├─ index.html
├─ koa-mock------------------------------------------基于koa实现的本地mock项目
│    ├─ api------------------------------------------将接口返回的json放在此目录，接口即为api/文件名
├─ package.json
├─ src-----------------------------------------------项目资源目录
│    ├─ App.vue--------------------------------------uni的主组件，所有页面都是在App.vue下进行切换的，是页面入口文件
│    ├─ api------------------------------------------API文件，用于集中管理接口
│    ├─ http-----------------------------------------axios拦截器，用于自定义请求和响应拦截
│    ├─ main.ts--------------------------------------入口文件，主要作用是初始化vue实例、定义全局组件
│    ├─ manifest.json--------------------------------应用的配置文件，用于指定应用的名称、图标、权限等
│    ├─ model----------------------------------------API接口用到的typescript模型，集中管理
│    ├─ pages----------------------------------------业务页面
│    ├─ pages.json-----------------------------------页面全局配置，决定页面文件的路径、窗口样式、原生的导航栏、底部的原生tabbar 等
│    ├─ router---------------------------------------路由配置，导航守卫
│    ├─ static---------------------------------------资源文件，图片等
│    ├─ store----------------------------------------pinia配置，持久化存储
│    │    ├─ index.ts
│    │    └─ persist.ts------------------------------持久化插件
│    ├─ types.d.ts-----------------------------------typescript的声明文件
│    ├─ uni.scss-------------------------------------全局scss变量，无需引入可以直接使用
│    └─ uni_modules----------------------------------uni-app插件
│           ├─ fant-mini-plus------------------------一款vue3组件库
│           └─ mp-html-------------------------------富文本插件
├─ tsconfig.json-------------------------------------TypeScript 编译器的配置文件
├─ versionrc.js--------------------------------------standard-version的配置文件
├─ vite.config.ts------------------------------------用于配置 vite 的编译选项
└─ yarn.lock-----------------------------------------Yarn生成的锁文件，它用于记录在使用Yarn安装包时确切的依赖关系和版本号
```

#### 开发步骤
- 安装依赖  
  - 安装前端项目依赖：`yarn `
  - 安装mock项目依赖：`yarn mock:install`
- 运行
  - 运行前端项目：`yarn dev:h5` （其他平台命令在uni-app官网或者package.json中查看）
  - 运行mock项目：`yarn mock:dev`
- 提交代码
  - 暂存准备提交的代码（举例添加当前目录下的所有文件到暂存区）：`git add .`
  - 提交：`yarn commit`，输入之后回车，我们可以选择提交类型。
- 发布版本（此处操作会使package.json中的version变更，可以此为版本号对显示）
  - 大版本升级1.0.0->2.0.0：`yarn release-major`
  - 中版本升级1.0.0->1.1.0：`yarn release-minor`
  - 小版本升级（一般是补丁）1.0.0->1.0.1：`yarn release-patch`
  - 注意：`执行完发布版本的命令之后需要将执行终端中打印的命令将生成的tag推送到git。此操作完成之后会自动生成版本发布日志，可以用于对外发布`


#### 开发提示  
- 使用typescript开发可以提高前后端交互的效率，提升前端项目的健壮性。在开发过程中需要做到尽量减少使用any。
小程序样式变量统一定义在uni.scss中，作为规范，若设计稿上使用的颜色存在于uni.scss中，则使用uni.scss中的变量，如不存在则需要沟通是否新增变量。

- 提交代码时，feat和fix类型的提交尽量将任务单号作为前缀方便追踪改动关联的需求，如：feat: TASK-1000 新增某个功能。
- css原子化，如果有需要可以自行引入[UnoCSS](https://unocss.dev/)，本项目暂时不会使用。
#### 主要命令

##### 安装依赖
```
yarn 
```
或者  
```
npm install
```
##### 安装mock依赖
```
yarn mock:install
```
或者  
```
npm run mock:install
```

##### 运行
```sh
# 运行到h5
yarn dev:h5
```
##### 运行mock服务
```sh
yarn mock:dev
```


##### Lint and fix
```
yarn lint
```

##### 提交代码
```sh
git add .

yarn commit
```

##### 发布版本


``` sh
# 大版本
yarn release-major  

# 中版本
yarn release-minor

# 小版本
yarn release-patch
```

##### 持续集成

```sh

# 上传到微信小程序
yarn upload:mp-weixin  

# 上传到支付宝小程序
yarn upload:mp-alipay

# 上传到钉钉小程序
yarn upload:mp-dingtalk
```


## 联系我

有不明白或者建议可以扫码交流
#### QQ群
<img  src="https://fant-mini-plus.top/img/qq.jpg
" width="220" height="auto">

#### 微信群
<img src="https://fant-mini-plus.top/img/weixin.png
" width="220" height="auto">




CHANGELOG.md
# 更新日志 


### [1.0.22](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.21...v1.0.22) (2023-07-07)


### ✨ Features | 新功能

* ✨ 更新uni-mini-router@0.1.3，修复前置导航守卫失效的问题 ([e52faed](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/e52faed8c3be49c5fd7f757549dc7ea510cecb7a))

### [1.0.21](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.20...v1.0.21) (2023-07-06)

### [1.0.20](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.19...v1.0.20) (2023-06-05)


### ✨ Features | 新功能

* ✨ 更新fant-mini-plus@0.1.7 支持Collapse异步更新、Table组件取消排序 ([dddb2fd](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/dddb2fd70d6cbf01fd0f36e45d589b2ad91af6d5))

### [1.0.19](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.18...v1.0.19) (2023-05-31)


### ✨ Features | 新功能

* ✨ 优化路由管理，支持路由表信息热更新 ([01b78c8](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/01b78c8e2b3c5dd8500ee5d8cee099589bd659ff))

### [1.0.18](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.17...v1.0.18) (2023-05-29)


### ✨ Features | 新功能

* ✨ 集成小程序端构建后支持 CI（持续集成）的插件 uni-mini-ci ([2c5caad](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/2c5caad7c0b9e6dab4d661808a21186f3bdbf7ec))

### [1.0.17](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.16...v1.0.17) (2023-05-24)


### ✨ Features | 新功能

* ✨ 更新fant-mini-plus@0.1.4 ([2b75b5b](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/2b75b5bf5e72974203a108361b47aa9802c2efd3))

### [1.0.16](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.15...v1.0.16) (2023-05-23)


### ✨ Features | 新功能

* ✨ 更新fant-mini-plus@0.1.2 ([fb10423](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/fb10423fe691a5dc5a57805cba318bfac7bc8e1c))


### ✏️ Documentation | 文档

* ✏️  更新readme，增加开发计划 ([61c9022](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/61c902255eceaa69eea4e32776ac1e26a0e74c4f))

### [1.0.15](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.14...v1.0.15) (2023-05-22)


### ✨ Features | 新功能

* ✨ 更新fant-mini-plus@0.1.0版本 ([34343b3](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/34343b329d21f63021f80b526e05c6ba4c94a713))

### [1.0.14](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.13...v1.0.14) (2023-05-18)


### ✨ Features | 新功能

* ✨ 更新fant-mini-plus@0.0.13 ([b69d807](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/b69d8077edeaf8d3f1bf52b2a9263ae9b58ba5ce))

### [1.0.13](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.12...v1.0.13) (2023-05-16)

### [1.0.12](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.11...v1.0.12) (2023-05-15)

### [1.0.11](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.10...v1.0.11) (2023-05-11)


### 🐛 Bug Fixes | Bug 修复

* 🐛 修复飞书小程序引入axios@1.x报错的问题 ([86d92a2](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/86d92a2d321f6dcb8a647981821dd553094b1794))


### ✏️ Documentation | 文档

* ✏️  增加联系方式 ([89d4c77](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/89d4c775fb32522514b017b2222e845875a40637))

### [1.0.10](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.9...v1.0.10) (2023-05-10)


### 🚀 Chore | 构建/工程依赖/工具

* 🚀 引入unplugin-auto-import支持自动按需引入第三库API ([913e23f](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/913e23f37dff0018ccbca6700a74d20b9651e851))

### [1.0.9](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.8...v1.0.9) (2023-05-08)


### Features

* ✨ 增加支持在导航守卫中改变跳转类型的示例 ([5c47eb5](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/5c47eb5b9a38e1d4439614e50958638d4932849a))

### [1.0.8](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.7...v1.0.8) (2023-05-06)

### [1.0.7](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.6...v1.0.7) (2023-04-27)


### Features

* ✨ 新增路由使用方法的演示页面 ([452da84](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/452da84984afbe0e279900a3b900701f1f784405))

### [1.0.6](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.5...v1.0.6) (2023-04-24)

### [1.0.5](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.4...v1.0.5) (2023-04-20)


### Features

* ✨ 已发出网络请求支持取消 ([0bd007b](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/0bd007bd25a514b44584536ea53f45c9a33af359))

### [1.0.4](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.3...v1.0.4) (2023-04-19)


### Bug Fixes

* 🐛 更新fant-axios-adapter修复接口请求header未携带的问题 ([905f784](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/905f7848f4e710a162e036695ae375300765d078))

### [1.0.3](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.2...v1.0.3) (2023-04-19)

### [1.0.2](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.1...v1.0.2) (2023-04-19)


### Features

* ✨ 更新fant-axios-adapter并移出一些无用依赖 ([8c0d182](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/8c0d182bebdbc3d8276503495a704c497d930265))

### [1.0.1](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/compare/v1.0.0...v1.0.1) (2023-04-19)

## 1.0.0 (2023-04-18)


### Features

* ✨ 引入fant-mini-plus，支持vue3、pinia持久化、axios、router等 ([fd182b1](https://gitee.com/fant-mini/uniapp-vue3-fant-ts/commit/fd182b1433ec95c7b491036bd1ac0a700a81715f))


vite.config.ts
/*
 * @Author: weisheng
 * @Date: 2023-03-21 22:49:24
 * @LastEditTime: 2023-05-10 10:04:00
 * @LastEditors: weisheng
 * @Description:
 * @FilePath: \uniapp-vue3-fant-ts\vite.config.ts
 * 记得注释
 */
import { defineConfig } from 'vite'
import TransformPages from 'uni-read-pages-vite'
import uni from '@dcloudio/vite-plugin-uni'
import AutoImport from 'unplugin-auto-import/vite'
export default defineConfig({
  base: './',
  plugins: [
    uni(),
    AutoImport({
      imports: [
        'vue',
        'uni-app',
        'pinia',
        {
          from: 'uni-mini-router',
          imports: ['createRouter', 'useRouter', 'useRoute']
        }
      ],
      dts: 'src/auto-imports.d.ts',
      dirs: ['src/store'],
      eslintrc: {
        enabled: true,
        globalsPropValue: true
      }
    })
  ],
  define: {
    ROUTES: new TransformPages().routes
  }
})
